### PR TITLE
[CRED-816] Define #post and #login. Add tests

### PR DESCRIPTION
**Pull request by alanwernicksf**

### CODE REVIEW

File: lib/billdotcom_integration/middleware/raise_failed_operation_error.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require 'faraday'
+
+module BillDotComIntegration
+  module Middleware
+    class RaiseOperationFailedError < Faraday::Middleware
+      ERROR_RESPONSE_STATUS = 1
```
Comment (jibrankalia-sf) [inline lib/billdotcom_integration/middleware/raise_failed_operation_error.rb]: Is this hardcoded from Bill.com?

File: lib/billdotcom_integration/middleware/raise_failed_operation_error.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+      def call(env)
+        @app.call(env).on_complete do |response|
+          return unless response.status == 200 && response.body
```
Comment (jibrankalia-sf) [inline lib/billdotcom_integration/middleware/raise_failed_operation_error.rb]: Is 200 the only valid status from bill.com?

File: spec/billdotcom_integration/rest_client_spec.rb [hunk 3]
```diff
@@ -23,6 +25,92 @@
...
+        stub_const(organization_id_const_name, "123456")
+      end
+
+      it 'should send a request to the login action' do
+        api_request = BillDotComIntegration::ApiRequest.new(BillDotComIntegration::Action::LOGIN)
+        api_request_full_path = URI::join(api_endpoint, api_request.path)
+        stub_request(:post, api_request_full_path).to_return(status: 200)
+
+        @rest_client.login
+        assert_requested :post, api_request_full_path
```
Comment (jibrankalia-sf) [inline spec/billdotcom_integration/rest_client_spec.rb]: Nice. First time I have seen assert_requested.

File: lib/billdotcom_integration/middleware/raise_failed_operation_error.rb (hunk 1 above)
Comment (alanwernicksf) [reply lib/billdotcom_integration/middleware/raise_failed_operation_error.rb]: Yeah :)

File: lib/billdotcom_integration/middleware/raise_failed_operation_error.rb (hunk 2 above)
Comment (alanwernicksf) [reply lib/billdotcom_integration/middleware/raise_failed_operation_error.rb]: For now yeah. I envision adding a middleware that catches 500 errors, etc. But 200 is the only valid one (even for invalid sessions)

**Pull request merged**